data/sea/10-mempool.h

```diff
@@ -1,131 +1,17 @@
-    pool->current_ptr = next->ptr;
-    pool->maximum_ptr = next->ptr + iblock_size;
-}
-
-/* This has to be a macro as when it's a function we end up with an extra
- * conditional because we need to return something to indicate whether we
- * succeeded or not. */
-#define TRY_ALLOC(fn_name)                                                                             \
-    void *ptr  = pool->current_ptr;                                                                    \
-    void *next = ptr + num_bytes;                                                                      \
-                                                                                                       \
-    if (next <= pool->maximum_ptr) {                                                                   \
-        ICICLE_WHEN_DEBUG (fprintf (stderr, #fn_name ": %p (allocated %zu bytes)\n", ptr, num_bytes)); \
-        pool->current_ptr = next;                                                                      \
-        return ptr;                                                                                    \
-    }
-
-static void * NOINLINE imempool_alloc_block (imempool_t *pool, size_t num_bytes)
-{
-    imempool_add_block (pool);
-
-    TRY_ALLOC (imempool_alloc_block);
-
-    /* Couldn't allocate even after adding a new block to the pool, we will
-     * never be able to service this request. */
-    fprintf (stderr, "oh my, someone's a bit greedy\n");
-    fprintf (stderr, "did you really need %zu bytes\n", num_bytes);
-    exit (1);
-}
-
-static void * INLINE imempool_alloc (imempool_t *pool, size_t num_bytes)
-{
-    TRY_ALLOC (imempool_alloc);
-
-    return imempool_alloc_block (pool, num_bytes);
-}
-
-/* Non-static functions cannot be inlined because of an old version of gcc
- * However we also need to be able to call this from Haskell */
-void * imempool_alloc_extern (imempool_t *pool, size_t num_bytes)
-{
-    return imempool_alloc(pool, num_bytes);
-}
-
-imempool_t * imempool_create ()
-{
-    imempool_t *pool = calloc(1, sizeof(imempool_t));
-
-#if ICICLE_DEBUG
-    fprintf (stderr, "imempool_create: %p\n", pool);
-#endif
-
-    imempool_add_block (pool);
-
-    return pool;
-}
-
-void imempool_debug_block_usage (imempool_t *pool)
-{
-    const iblock_t *block = pool->last;
+    const anemone_block_t *block = pool->last;
     uint64_t size = 0;
     uint64_t blocks = 0;
     while (block != 0) {
-      size += iblock_size;
+      size += anemone_block_size;
       blocks++;
       block = block->prev;
     }
-    fprintf (stderr, "imempool_debug_block_usage: %" PRIu64 " blocks, %" PRIu64 " bytes\n", blocks, size);
-}
-
-void imempool_free (imempool_t *pool)
-{
-#if ICICLE_DEBUG
-    fprintf (stderr, "imempool_free: %p\n", pool);
-    imempool_debug_block_usage (pool);
-#endif
-
-    iblock_free (pool->last);
-    free(pool);
+    fprintf (stderr, "anemone_mempool_debug_block_usage: %" PRIu64 " blocks, %" PRIu64 " bytes\n", blocks, size);
```

Comment:
wait. I think this function (or a similar one) is also defined in anemone_mempool.c, and I was getting them confused